examples-c/ethereum_bls_signatures.c

```diff
@@ -49,7 +49,49 @@ int main(){
     printf("Signature verification failure: status %d - %s\n", bls_status, ctt_eth_bls_status_to_string(bls_status));
     exit(1);
   }
-
   printf("Example BLS signature/verification protocol completed successfully\n");
+
+
+  // ------------------------------
+  // Batch verification
+  // ------------------------------
+
+  // try to use batch verify; We just reuse the data from above 3 times
+  const ctt_eth_bls_pubkey pkeys[3] = { pubkey, pubkey, pubkey };
+  ctt_span messages[3] = { // already hashed message, reuse 3 times
+      { message, 32 },
+      { message, 32 },
+      { message, 32 }
+  };
+  const ctt_eth_bls_signature sigs[3] = { sig, sig, sig };
+  byte srb[32] = {0}; // just a bunch of zeros as random "secure" bytes.
```

Comment:
a bunch of 7, 8 or anything is better, because internally we multiply the randomness with the inputs and multiplying with 0 gives 0, and pairing of a zero input always gives 1.

To prevent that we re-hash here
https://github.com/mratsim/constantine/blob/2a2bf9c0c777d283265ce66632b43c9a0f9e1cb7/constantine/signatures/bls_signatures.nim#L404-L411

But I think it is a good occasion to show sysrand